src/core/Akka/Event/Logging.cs

```diff
@@ -64,18 +64,25 @@ public static LogSource Create(object o, ActorSystem system)
                     return new LogSource(FromActorRef(actorRef, system), SourceType(actorRef));
                 case string str:
                     return new LogSource(FromString(str, system), SourceType(str));
-                case System.Type t:
+                case Type t:
                     return new LogSource(FromType(t, system), t);
+                case LogSource logSource:
+                    return logSource; // if someone's already created a LogSource, just use it
```

Comment:
If someone calls `LogSource.Create` with a ... `LogSource`... as input, just use the `LogSource`. This was "eating" user-defined `LogSource` names before.